src/main/java/org/jboss/dmr/ValueExpressionResolver.java

```diff
@@ -194,14 +195,40 @@ protected String resolvePart(String name) {
         } else if (":".equals(name)) {
             return File.pathSeparator;
         }
-        // First check for system property, then env variable
+        // First check for system property, then env variables
         String val = System.getProperty(name);
+
+        if (val == null) {
+            // See if an env var is defined
+            String envVar = replaceNonAlphanumericByUnderscores(name);
+            envVar = envVar.toUpperCase(Locale.ENGLISH);
```

Comment:
You might as well have `replaceNonAlphanumericByUnderscores` use `toUpperCase` on each letter and then you can skip this bit.  `Character.toUpperCase` will use root-locale rules.